### DISCUSSION TITLE
Rule proposal: no-input-native

### DISCUSSION THREAD

**Discussion by mattlewis92** [angular-eslint/angular-eslint#824] [New Rule Proposals]
For this rule it would work exactly the same as [`no-output-native`](https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin/src/rules/no-output-native.ts) but disallow naming any inputs the same as a html attribute e.g. `title`, `class`, `draggable` etc. The full list can be found [here](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes). More than happy to work on this myself if it's accepted 😄

**Comment by rafaelss95** [comment]
[It looks like it was also proposed in Codelyzer](https://github.com/mgechev/codelyzer/issues/661) and is related to https://github.com/angular/angular/issues/24376.

This is something I would love to see here too, however, one case that still intrigues me is how to express the `id` (and keep its purpose) in a reusable component, such as those used by the Material Team, which has a input from most (if not all) of its components. Perhaps we should have an option to allow the user to pass an ignore pattern?

Also, what about the Angular "native" inputs, like `form-*` (`formControl`, `formGroup`, etc.), `routerLink` that can cause some conflicts?

**Comment by mattlewis92** [comment]
> Perhaps we should have an option to allow the user to pass an ignore pattern?

I think that's a good idea!


> Also, what about the Angular "native" inputs, like `form-*` (`formControl`, `formGroup`, etc.), `routerLink` that can cause some conflicts?

I would probably say it's out of scope for this rule, it's more of a general problem if you name a directive with the same selector as an input on a component. It's probably something more suited to the angular compiler than a lint rule.

**Comment by makeAdance** [comment]
What's the current state? Our development team definitely would like to have this rule as well. :)

**Comment by LukasMachetanz** [comment]
What's the current state? Is this planned to be implemented?

**Comment by JamesHenry** [comment]
I think it makes sense to add this rule, so anybody is welcome to submit a PR
